.github/workflows/upgrade-ha.yaml

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 Defense Unicorns
+# SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+
+name: UDS Core Install/Upgrade HA Test
+
+on:
+  schedule:
+    # Runs every morning at 2:00 AM UTC
+    - cron: '0 2 * * *'
+
+permissions:
+  contents: read
+  id-token: write
+  packages: read
+
+jobs:
+  uds-core-ha-upgrade-nightly:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        flavor: [upstream, registry1, unicorn]
```

Comment:
Maybe this falls into the bucket of "testing too much", but would we also want to matrix this with install/upgrade (to test both)? Open to either here - and the upgrade is probably the more important one to add initially.